test/e2e_env/universal/meshretry/http.go

```diff
@@ -71,32 +86,109 @@ spec:
 				universal.Cluster, "demo-client", "test-server.mesh",
 			)
 			g.Expect(err).ToNot(HaveOccurred())
-		}).Should(Succeed())
-		Consistently(func(g Gomega) {
-			// --max-time 8 to wait for 8 seconds to beat the default 5s connect timeout
+		}, "10s", "100ms", MustPassRepeatedly(5)).Should(Succeed())
+
+		By("Adding a MeshFaultInjection for test-server")
+		Expect(universal.Cluster.Install(YamlUniversal(meshFaultInjection))).To(Succeed())
+
+		By("Check some errors happen")
+		Eventually(func(g Gomega) {
+			_, err := client.CollectEchoResponse(
+				universal.Cluster, "demo-client", "test-server.mesh",
+			)
+			g.Expect(err).To(HaveOccurred())
```

Comment:
What error are we getting here? Can we check more, maybe we can see the status code is 500. It would be nice to rule out non-relevant errors here (like connectivity or DNS)